packages/fast-components-styles-msft/src/utilities/color/palette.ts

```diff
@@ -251,24 +251,74 @@ export function swatchByContrast(referenceColor: string | SwatchResolver) {
                             designSystem
                         );
                         const length: number = sourcePalette.length;
-                        let index: number = initialSearchIndex;
 
-                        while (
-                            inRange(index + direction, 0, length) &&
-                            inRange(index, 0, length) &&
-                            !contrastCondition(contrast(color, sourcePalette[index]))
-                        ) {
-                            index = index + direction;
+                        function contrastSeachCondition(
+                            valueToCheckAgainst: Swatch
+                        ): boolean {
+                            return contrastCondition(
+                                contrast(color, valueToCheckAgainst)
+                            );
                         }
 
-                        return sourcePalette[index];
+                        let constrainedSourcePalette: Palette = [];
+
+                        if (direction === 1) {
+                            constrainedSourcePalette = sourcePalette.slice(
+                                initialSearchIndex,
+                                length
+                            );
+                        } else {
+                            constrainedSourcePalette = sourcePalette.slice(
+                                0,
+                                initialSearchIndex + 1
+                            );
+                            // reverse the palette array when the direction that
```

Comment:
Do you need to slice the palette now that you're simply tracking an index?

Reply:
I'll try it without, I was attempting to do the binary search on the assumption that 0 is always the searching index and it always moves in a positive direction (aka darker).